### DISCUSSION TITLE
iteration vs milestone

### DISCUSSION THREAD

**Discussion by mesqueeb** [community/community#11832] [Projects and Issues]
Previously we were using milestones we set up every 2 weeks to decide which tickets in our repo go into which sprint.
Now there's "iterations" which seem to be recommended for 2 week sprints.

What's the difference and pros/cons to each of these?

**Comment by sthakkar** [comment]
I would also like to understand this and see if there is a recommended best way to leverage iterations vs milestones.

Is one supposed to be considered shorter term than the other?

**Reply by silopolis** [reply]
I believe iterations should be (possibly much) shorter than milestones, as:
* iterations tend to be as short as possible and consume user stories or even tasks
* milestones correspond to (much) higher level project splitting/stepping and potentially group several epics

**Comment by d3rp3tt3** [comment]
One significant difference is that an iteration may be used across multiple repos via projects. A milestone, however, applies directly to the metadata of an issue and can be applied to issues in one repo only.

**Reply by silopolis** [reply]
This makes me think milestones should be moved, where they belong, to Projects, as they really are a project management object, that may need to be shared between repositories.

**Reply by theking2** [reply]
> This makes me think milestones should be moved, where they belong, to Projects, as they really are a project management object, that may need to be shared between repositories.

My thoughts exactly. I like the one-stop show in github clients (Github Desktop, Github Pull Requests) and it would be great if they are aware of GitHub Projects as well in some shape or form.

**Comment by yepes** [comment]
A clear explanation in the docs about when to use each and a few examples would be nice.... right now, pretty unusable
